pkg/datapath/linux/routing/routing_test.go

```diff
@@ -209,6 +209,16 @@ func listRulesAndRoutes(c *C, family int) ([]netlink.Rule, []netlink.Route) {
 // used as a harness in this test. This function returns a function which can
 // be used to remove the device for cleanup purposes.
 func createDummyDevice(c *C, macAddr mac.MAC) func() {
+	links, err := netlink.LinkList()
+	c.Assert(err, IsNil)
+
+	for _, link := range links {
+		if link.Attrs().HardwareAddr.String() == macAddr.String() {
+			c.Logf("Found device with identical mac addr: %s", macAddr.String())
+			c.FailNow()
+		}
+	}
```

Comment:
It might be worth moving this to a separate function to reduce duplication and make it easier to read.